src/Core/src/Handlers/DatePicker/DatePickerHandler.iOS.cs

```diff
@@ -39,18 +33,18 @@ protected override MauiDatePicker CreatePlatformView()
 			return platformDatePicker;
 		}
 
-		internal UIDatePicker? DatePickerDialog { get { return _picker; } }
+		internal UIDatePicker? DatePickerDialog { get { return PlatformView?.InputView as UIDatePicker; } }
 
 		internal bool UpdateImmediately { get; set; }
 
 		protected override void ConnectHandler(MauiDatePicker platformView)
 		{
-			if (_picker is UIDatePicker picker)
-			{
-				picker.EditingDidBegin += OnStarted;
-				picker.EditingDidEnd += OnEnded;
-				picker.ValueChanged += OnValueChanged;
+			platformView.InputViewEditingDidBegin += OnStarted;
```

Comment:
These are all exposed as weakevents inside the MauiDatePicker now. This way these events are only part of the platformView opposed to a subview of the platformview